src/no/runsafe/nchannels/database/ChannelMembershipRepository.java

```diff
@@ -44,19 +48,20 @@ public ISchemaUpdate getSchemaUpdateQueries()
 		return update;
 	}
 
+	public Boolean isEmptyChannel(String channel)
+	{
+		List<String> stringIds = database.queryStrings("SELECT player FROM nchannel_members WHERE channel=?", channel);
+		return stringIds.isEmpty();
+	}
+
 	public List<String> getPlayerChannels(IPlayer player)
 	{
 		return database.queryStrings("SELECT channel FROM nchannel_members WHERE player=?", player.getUniqueId().toString());
 	}
 
-	public List<UUID> getChannelPlayers(String channel)
+	public List<IPlayer> getChannelPlayers(String channel)
 	{
-		List<String> stringIds = database.queryStrings("SELECT player FROM nchannel_members WHERE channel=?", channel);
-		List<UUID> playerIds = new ArrayList<UUID>();
-		for(String playerStringId : stringIds)
-			playerIds.add(UUID.fromString(playerStringId));
-
-		return playerIds;
+		return playerProvider.getPlayersByIDs(database.queryStrings("SELECT player FROM nchannel_members WHERE channel=?", channel));
```

Comment:
I meant to add a new method.. but you could change that one to be smart about what it treats values as.. it's going to be either username or uuid and it should all become uuid over time anyway.